packages/block-editor/src/store/reducer.js

```diff
@@ -1277,6 +1277,11 @@ export const blockListSettings = ( state = {}, action ) => {
  * @return {string} Updated state.
  */
 export function isNavigationMode( state = false, action ) {
+	// Let inserting block always trigger Edit mode.
+	if ( action.type === 'INSERT_BLOCKS' ) {
+		return false;
```

Comment:
@youknowriad @MarcoZehe Curious to know what you think about this. Edit mode will be enabled if you insert a block, as most likely you'll be wanting to edit the block, even if you came from navigation mode. I can also omit this if it doesn't make sense.

Reply:
I actually think it does make sense, since inserting the block usually means that you will want to edit it right away. The mobile app behaves like this, too, once you insert a block, you are usually thrown into text editing mode if it's a block you can edit like this.

Reply:
Ok, let's do it. We can always revert this specific part if we change our mind.